src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.Instance.cs

```diff
@@ -0,0 +1,135 @@
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EnvDTE;
+using Microsoft.VisualStudio.ProjectSystem.Refactor;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Refactor
+{
+    internal sealed partial class VisualStudioRefactorNotifyService
+    {
+        internal class Instance : IMultiLifetimeInstance, IRefactorNotifyService
+        {
+            private readonly IVsService<SDTE, DTE> _dte;
+            private readonly IVsService<SVsSolution, IVsSolution> _solutionService;
+
+            public Instance(IVsService<SDTE, DTE> dte, IVsService<SVsSolution, IVsSolution> solutionService)
+            {
+                _dte = dte;
+                _solutionService = solutionService;
+            }
+
+            public async Task<bool> TryOnBeforeGlobalSymbolRenamedAsync(string projectPath, IEnumerable<string> filePaths, string rqName, string newName)
+            {
+                IVsHierarchy projectHierarchy = await GetProjectHierarchy(projectPath);
+                if (projectHierarchy == null)
+                {
+                    return false;
+                }
+
+                if (!(projectHierarchy is IVsHierarchyRefactorNotify refactorNotify))
+                {
+                    return false;
+                }
+
+                var ids = GetIdsForFiles(projectHierarchy, filePaths).ToArray();
+
+                // call OnBeforeGlobalSymbolRenamed
+                if (refactorNotify.OnBeforeGlobalSymbolRenamed(cItemsAffected: (uint)ids.Length,
+                                                               rgItemsAffected: ids,
+                                                               cRQNames: (uint)1,
+                                                               rglpszRQName: new[] { rqName },
+                                                               lpszNewName: newName,
+                                                               promptContinueOnFail: 1) != HResult.OK)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            public async Task<bool> TryOnAfterGlobalSymbolRenamedAsync(string projectPath, IEnumerable<string> filePaths, string rqName, string newName)
+            {
+                IVsHierarchy projectHierarchy = await GetProjectHierarchy(projectPath);
+                if (projectHierarchy == null)
+                {
+                    return false;
+                }
+
+                if (!(projectHierarchy is IVsHierarchyRefactorNotify refactorNotify))
+                {
+                    return false;
+                }
+
+                var ids = GetIdsForFiles(projectHierarchy, filePaths).ToArray();
+
+                // call OnBeforeGlobalSymbolRenamed
```

Comment:
Nit: Comment is wrong

Reply:
I'd vote for removing this comment and the one above from which it was presumably copied. They're stating the obvious.